### ISSUE TITLE
remote-shell-cmd does not close ssh connections

### ISSUE THREAD

**Issue by knagy** [StackStorm/st2#3274]
Hi,

If I run a `core.remote` action an ssh connection opens and it does not close after the action finishes successfully. Running the same action again and again opens new connections until the remote server refuses to accept more.

A simple bash one-liner to reproduce this:

```
for i in {1..1000}; do st2 run core.remote hosts=127.0.0.1 whoami; done
```

And the result:

```
$ netstat -a | grep "tcp \+0 \+0 \+localhost:ssh" | wc -l
1000
```

I tried it on a fresh Ubuntu 16.04 machine with a fresh install of StackStorm 2.2.0.

BR,
k.

**Comment by Kami** [comment]
Thanks for reporting this.

Can you please post more details - which version of StackStorm you are running and which version of paramiko library is used (`/opt/stackstorm/bin/st2/pip show paramiko`).

We haven't noticed anything like this in the past, but we recently upgraded to latest version of paramiko so it's possible some default behavior is changed or maybe it's related to some setting in the user / global SSH config file .

To rule out some SSH config related keep alive setting keeping the connections open, can you please also run something like this:

```bash
for i in {1..1000}; do st2 run core.remote hosts=127.0.0.1 whoami; sleep 20; done
```

**Comment by Kami** [comment]
I was able to reproduce the issue - I'm looking into a fix.

Thanks again for reporting this.

**Comment by Kami** [comment]
Fix in #3275.

Thanks again for reporting this.

**Comment by knagy** [comment]
Thanks for the quick response. I tried it out on my test environment and works perfectly.

**Comment by Kami** [comment]
Thanks for confirming.

I will wait for @lakshmi-kannan on his review and if everything looks OK, I will go ahead and merge it.

This change will then be available in v2.2.1 or v2.3.0 (we are still not decided if we will do v2.2.1 before v2.3.0 or not).
